Checkstyle should use `dependencyConvergence` maven-enforcer rule

Noticed at https://app.circleci.com/pipelines/github/checkstyle/checkstyle/15272/workflows/7e77c653-dfbf-4524-9386-3f044e201328/jobs/180000, our build in hibernate is failing, but ours does not. This is because hibernate uses the [dependencyConvergence](https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html#:~:text=Dependency%20Convergence,-This%20rule%20requires&text=If%20a%20project%20has%20two,cause%20a%20build%20to%20fail.&text=And%20this%20will%20succeed.) rule. We should as well.

Upon activation, we have one violation:
```
[WARNING] 
Dependency convergence error for org.slf4j:slf4j-api:jar:1.7.32:compile paths to dependency are:
+-com.puppycrawl.tools:checkstyle:jar:10.3.4-SNAPSHOT
  +-org.reflections:reflections:jar:0.10.2:compile
    +-org.slf4j:slf4j-api:jar:1.7.32:compile
and
+-com.puppycrawl.tools:checkstyle:jar:10.3.4-SNAPSHOT
  +-net.sf.saxon:Saxon-HE:jar:11.4:compile
    +-org.xmlresolver:xmlresolver:jar:4.4.3:compile
      +-org.apache.httpcomponents.client5:httpclient5:jar:5.1.3:runtime
        +-org.slf4j:slf4j-api:jar:1.7.25:runtime

[ERROR] Rule 2: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
```

Not only will this rule help to catch issues in our own project, but it will also help out users by not making them deal with dependency issues. See https://github.com/hibernate/hibernate-search/blob/2792ce59de42d11c644934ab57b4cb689ef47e96/pom.xml#L2022

--- a/pom.xml
+++ b/pom.xml
@@ -381,6 +381,10 @@
           <groupId>xml-apis</groupId>
           <artifactId>xml-apis</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
   </dependencies>
@@ -1309,6 +1313,7 @@
                 <requireMavenVersion>
                   <version>3.3.9</version>
                 </requireMavenVersion>
+                <dependencyConvergence/>
                 <!-- we can not use this as it require same version for all dependencies -->
                 <!--
                 <enforceBytecodeVersion>

